flight/Modules/Stabilization/stabilization.c

```diff
@@ -219,6 +228,25 @@ static void stabilizationTask(void* parameters)
 		dT = PIOS_DELAY_DiffuS(timeval) * 1.0e-6f;
 		timeval = PIOS_DELAY_GetRaw();
 		
+		// exponential moving averaging (EMA) of dT to reduce jitter; ~200points
+		// to have more or less equivalent noise reduction to a normal N point moving averaging:  alpha = 2 / (N + 1)
+		// run it only at the beginning for the first samples, to reduce CPU load, and the value should converge to a constant value
+		if (dT_filter_iteration-- > 2) {
+			dT_filtered = 0.01f * dT + (1.0f - 0.01f) * dT_filtered;
+
+			if (gyro_filter_enabled  == GYRO_LPF_DISABLED)
+				gyro_alpha = 0;   // Gyro LPF should be disabled
+			else
+				gyro_alpha = expf(-2.0f * (float)(M_PI) * settings.GyroCutoff * dT_filtered);
+		}
+
+
+		// new calculation of gyro_alpha when cutoff frequency has changed, and dt_filtered is already calculated
+		if ( (gyro_filter_enabled  == GYRO_LPF_ENABLED_UPDATE) && (dT_filter_iteration == 1) ) {
```

Comment:
I suggest if ( ( gyro_filter_enable == GYRO_LPF_ENABLED_UPDATE) || (dT_filter_iteration > 0)).    Then the computation of gyro_alpha can be removed from above.  Since you have a flag for GYRO_LPF_ENABLED, the conditional to set the gyro_alpha to 0 is not needed; the code doesn't look at it in the DISABLED state.